### DISCUSSION TITLE
CARLA 0.9.14, make PythonAPI error plz help

### DISCUSSION THREAD

**Discussion by kim-hongtae** [carla-simulator/carla#6165] [Q&A] [answered]
CARLA 0.9.14
ubuntu 20.04

I'm getting an error in the make PythonAPI step, please let me know how to solve it.

Check for working CXX compiler: /usr/bin/clang++-10 -- broken
CMake Error at /usr/share/cmake-3.16/Modules/CMakeTestCXXCompiler.cmake:53 (message):
  The C++ compiler

    "/usr/bin/clang++-10"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /home/hongtae/carla/Build/llvm-8.0-build/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/usr/bin/ninja cmTC_c25ed && [1/2] Building CXX object CMakeFiles/cmTC_c25ed.dir/testCXXCompiler.cxx.o
    [2/2] Linking CXX executable cmTC_c25ed
    FAILED: cmTC_c25ed 
    : && /usr/bin/clang++-10     CMakeFiles/cmTC_c25ed.dir/testCXXCompiler.cxx.o  -o cmTC_c25ed   && :
    /usr/bin/ld: cannot find -lstdc++
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:39 (project)

-- Configuring incomplete, errors occurred!
See also "/home/hongtae/carla/Build/llvm-8.0-build/CMakeFiles/CMakeOutput.log".
See also "/home/hongtae/carla/Build/llvm-8.0-build/CMakeFiles/CMakeError.log".
make: *** [Util/BuildTools/Linux.mk:137: setup] Error 1

**Comment by jemishGhoghari** [comment]
Hello @HoongTae 
Have you found the solution for your Problem? because I am having the same error while running "make PythonAPI".

could you please help me, if you have figured out the problem?

Regards
Jemish

**Reply by kim-hongtae** [reply, marked as answer]
Hi @jemishGhoghari 

I've figured out a workaround though not a solution.
First of all, to run the 'make PythonAPI' command, you need to compile it with clang-8.
However, I found out that clang-8 cannot be installed on Ubuntu 20.04.
So I mounted carla folder with docker and created a docker container with ubuntu 18.04 environment.
Then I solved it by running compile inside the container!
I hope it works well.

Hongtae
